src/node_process_methods.cc

```diff
@@ -130,6 +130,71 @@ static void CPUUsage(const FunctionCallbackInfo<Value>& args) {
   fields[1] = MICROS_PER_SEC * rusage.ru_stime.tv_sec + rusage.ru_stime.tv_usec;
 }
 
+// ThreadCPUUsage use system dependent implementation to get the current thread
+// cpu times:
+//
+// - On Unix, it uses getrusage(2) with RUSAGE_THREAD.
+// - On macOS, it uses thread_info(2) with THREAD_BASIC_INFO.
+// - On Windows, it uses GetThreadTimes(2).
+//
+// Returns those values as Float64 microseconds in the elements of the array
+// passed to the function.
+static void ThreadCPUUsage(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  // Get the double array pointer from the Float64Array argument.
+  Local<ArrayBuffer> ab = get_fields_array_buffer(args, 0, 2);
+  double* fields = static_cast<double*>(ab->Data());
+
+#ifdef __linux__
+  clockid_t clockid;
+  pthread_getcpuclockid(pthread_self(), &clockid);
+
+  struct rusage usage;
+
+  if (getrusage(RUSAGE_THREAD, &usage) == -1) {
```

Comment:
Could you add a comment to each of these mysterious functions and explain what they do? It might be tricky for someone else who comes across this code in a few years.

Reply:
Sure way!

Reply:
Done